.github/workflows/gh-pages.yml

```
@@ -82,3 +82,34 @@ jobs:
       - name: Push
         run: |-
           git push origin gh-pages:gh-pages
+
+  benchmark:
+    name: Benchmark
+    needs: publish
+    if: github.event_name == 'push' || github.event_name == 'workflow_dispatch'
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Node
+        uses: actions/setup-node@v3
+        with:
+          cache: yarn
+          node-version: '16'
+      - name: Install Dependencies
+        run: |-
+          yarn install --frozen-lockfile
+          yarn build
+      - name: Run Benchmark
+        run: |-
+          cd packages/@jsii/benchmarks
+          yarn bench | tee output.txt
+      - name: Store benchmark result
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          name: JSII Benchmark
+          tool: 'benchmarkjs'
+          output-file-path: packages/@jsii/benchmarks/output.txt
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          auto-push: true
```

Instead of `cd`, you should use the `working-directory` property.

I'd also emit the results out-of-tree because it's cleaner.

```suggestion
      - name: Run Benchmark
        run: |-
          yarn bench | tee ${{ runner.temp }}/bench-output.txt
        working-directory: packages/@jsii/benchmarks
      - name: Store benchmark result
        uses: benchmark-action/github-action-benchmark@v1
        with:
          name: JSII Benchmark
          tool: 'benchmarkjs'
          output-file-path: ${{ runner.temp }}/bench-output.txt
          github-token: ${{ secrets.GITHUB_TOKEN }}
          auto-push: true
```